tests/test_project.py

```diff
@@ -2313,6 +2314,236 @@ def test_main_submit(self, monkeypatch):
+        else:
+            self.call_subcmd(f"run -o {operation_name} -j {job.id}")
+
+        if job.sp.raise_exception:
+            assert not get_job_doc_value(self.keys[2])
+        else:
+            assert get_job_doc_value(self.keys[2])
+
+    def test_fail(self, project, job, operation_name):
+        get_job_doc_value = self._get_job_doc_key(job, operation_name)
+
+        assert get_job_doc_value(self.keys[3]) is None
+
+        if job.sp.raise_exception:
+            with pytest.raises(subprocess.CalledProcessError):
+                self.call_subcmd(f"run -o {operation_name} -j {job.id}")
+        else:
+            self.call_subcmd(f"run -o {operation_name} -j {job.id}")
+
+        if job.sp.raise_exception:
+            assert get_job_doc_value(self.keys[3])[0]
+            assert get_job_doc_value(self.keys[3])[1] == self.error_message
+        else:
+            assert get_job_doc_value(self.keys[3]) is None
+
+
+class TestHooksCmd(TestHooksBase):
+    # Tests hook decorators for a job operation with the @cmd decorator
+    error_message = 42
+
+    @pytest.fixture(params=["base_cmd"])
+    def operation_name(self, request):
+        return request.param
+
+
+class TestHooksInstallSetUp(TestHooksSetUp):
+    entrypoint = dict(
+        path=os.path.realpath(
+            os.path.join(os.path.dirname(__file__), "define_hooks_install.py")
+        )
+    )
+
+
+class TestHooksInstallBase(TestHooksBase, TestHooksInstallSetUp):
+    # Tests project-wide hooks on job operations with and without operation level hooks
+
+    # Check job document for keys from installed, project-wide hooks
+    keys = [
+        "installed_start",
+        "installed_finish",
+        "installed_success",
+        "installed_fail",
+    ]
+
+    @pytest.fixture(params=["base", "base_no_decorators"])
+    def operation_name(self, request):
+        return request.param
+
+
+class TestHooksInstallCmd(TestHooksCmd, TestHooksInstallSetUp):
+    # Tests project-wide hooks on job operations with the @cmd decorator.
+    # Job operations are with or without operation level hooks
+
+    # Check job document for keys from installed, project-wide hooks
+    keys = [
+        "installed_start",
+        "installed_finish",
+        "installed_success",
+        "installed_fail",
+    ]
+
+    @pytest.fixture(params=["base_cmd", "base_cmd_no_decorators"])
+    def operation_name(self, request):
+        return request.param
+
+
+class TestHooksInstallWithDecorators(TestHooksBase, TestHooksInstallSetUp):
+    # Tests if project-wide hooks interfere with operation level hooks
+    @pytest.fixture(params=["base"])
+    def operation_name(self, request):
+        return request.param
+
+
+class TestHooksInstallCmdWithDecorators(TestHooksCmd, TestHooksInstallSetUp):
+    # Tests if project-wide hooks interfere with operation level hooks
+    # in job operations with the @cmd decorator
+    @pytest.fixture()
+    def operation_name(self):
```

Comment:
You are inconsistent on whether you parametrize single operation fixtures or use a blank signature. Pick one style and use that for consistency. 
```suggestion
    @pytest.fixture
    def operation_name(self):
```

Reply:
@b-butler I don't see any inconsistency from the changeset in this PR. I see that `test_aggregates.py` uses no parentheses here, but everywhere else in signac-flow (including this PR) uses `fixture()` as far as I can tell. I would recommend making a separate PR to fix `test_aggregates.py` if you would like. Let me know if I'm missing something.